collect_app/src/main/java/org/odk/collect/android/projects/ProjectDeleter.kt

```diff
@@ -2,29 +2,41 @@ package org.odk.collect.android.projects
 
 import org.odk.collect.android.backgroundwork.FormUpdateScheduler
 import org.odk.collect.android.backgroundwork.InstanceSubmitScheduler
-import org.odk.collect.projects.Project.Saved
+import org.odk.collect.forms.instances.InstancesRepository
+import org.odk.collect.projects.Project
 import org.odk.collect.projects.ProjectsRepository
 
 class ProjectDeleter(
     private val projectsRepository: ProjectsRepository,
     private val currentProjectProvider: CurrentProjectProvider,
     private val formUpdateScheduler: FormUpdateScheduler,
-    private val instanceSubmitScheduler: InstanceSubmitScheduler
+    private val instanceSubmitScheduler: InstanceSubmitScheduler,
+    private val instancesRepository: InstancesRepository
 ) {
-    fun deleteCurrentProject(): Saved? {
-        val currentProject = currentProjectProvider.getCurrentProject()
+    fun deleteCurrentProject(): DeleteProjectResult {
+        if (instancesRepository.allUnsent.isNotEmpty()) {
+            return DeleteProjectResult.UnsentInstances
+        } else {
+            val currentProject = currentProjectProvider.getCurrentProject()
 
-        formUpdateScheduler.cancelUpdates(currentProject.uuid)
-        instanceSubmitScheduler.cancelSubmit(currentProject.uuid)
+            formUpdateScheduler.cancelUpdates(currentProject.uuid)
+            instanceSubmitScheduler.cancelSubmit(currentProject.uuid)
 
-        projectsRepository.delete(currentProject.uuid)
+            projectsRepository.delete(currentProject.uuid)
 
-        return if (projectsRepository.getAll().isNotEmpty()) {
-            val newProject = projectsRepository.getAll()[0]
-            currentProjectProvider.setCurrentProject(newProject.uuid)
-            newProject
-        } else {
-            null
+            return if (projectsRepository.getAll().isNotEmpty()) {
+                val newProject = projectsRepository.getAll()[0]
+                currentProjectProvider.setCurrentProject(newProject.uuid)
+                DeleteProjectResult.DeletedSuccessfully(newProject)
+            } else {
+                DeleteProjectResult.DeletedSuccessfully(null)
+            }
         }
     }
 }
+
+sealed class DeleteProjectResult {
+    object UnsentInstances : DeleteProjectResult()
+
+    data class DeletedSuccessfully(val project: Project.Saved?) : DeleteProjectResult()
```

Comment:
Lets name this field `newCurrentProject` just to make it clearer which project is being returned.

Reply:
Done.